common/namer/namer.go

```diff
@@ -55,3 +61,12 @@ func (n Namer) DisplayName(parts ...pulumi.StringInput) pulumi.StringInput {
 		return n.ResourceName(strArgs...)
 	}).(pulumi.StringOutput)
 }
+
+// HashedName return the md5 32-chars hex pulumi.StringInput of `parts` prefixed
+// with the namer prefix
+func (n Namer) HashedName(parts ...pulumi.StringInput) pulumi.StringInput {
+	return n.DisplayName(parts...).ToStringOutput().ApplyT(func(name string) string {
```

Comment:
I'm not sure this is the logic we want as we're loosing info about stack name. We should perhaps just review the naming, size of the stack name and then generate something like `<stack-name|16>-<hash|16>`.

Reply:
What if `stack-name` is > 32 chars ?

Reply:
Unfortunately not all objects get tags (not all objects get names, even). We should review the naming patterns to understand the size of each piece but it's probably reasonable to limit the stack name to, say `16` chars

Reply:
There would be a higher risk of collisions, as we cannot control where the unique part comes from. The disadvantage here is that we don't know which username an LB belongs to, at first sight, right ?

Reply:
What about `{username.split(".")[0][0]}.{username.split(".")[1]}-{hash[:remaining_length]}` and we use the current hash, with the stack name included ?

Reply:
This would give something like `p.ducolin-1234BAD6` - for 32 chars total. We can remove the point.

Reply:
That's the logic used in `datadog-agent` already: https://github.com/DataDog/datadog-agent/blob/39703bcc5900903d0eeac29f6e727dae00544710/test/new-e2e/runner/local-profile.go#L31-L58

The code inside `test-infra` should not try to do things with stack name, except validating the length if we want to do that.

Reply:
Tried with your suggestion and we I got something like `paola-ducolin-aw6727d0b5bb5b88da`

Reply:
Updated !

Reply:
Do we need the `HashedName` function then?

Reply:
As the goal of object display names is to be as explicit as possible for users,
(if the goal was only to be unique, we could just not specify it and let pulumi generate a unique name for us)
wouldn’t it be valuable to try to keep the current untruncated concatenation of full stack name and full object name when it fits in the object name max size and switch to the truncated + hash name only when the full name doesn’t fit ?